pagefaultgames/pokerogue
# [Bug] IV scanner legacy text update

## What are the changes the user will see?
For the IV scanner's text colour, when the stat is worse than what you have is currently, it shows up in white text. This is fine in the standard UI theme, but for the legacy UI theme, since the IV scanner's background is also white, it makes it hard to see. This PR fixes the text to be not white for the legacy UI theme

## Why am I making these changes?
Someone brought it up in discord; here we are

## What are the changes from a developer perspective?
The old line of code that would get the IV descriptions would base the colour off of `TextStyle.SUMMARY`, but within the function that gets the actual colour (getTextColor), there's no alternative colour option for the UI theme. I had a look around and saw that the names of the stats were working well using `TextStyle.WINDOW`; this is because `TextStyle.WINDOW` has an allowance for different UI themes. You can see the getTextColor function [here](https://github.com/pagefaultgames/pokerogue/blob/main/src/ui/text.ts#L229)

I also removed a line of code that was commented out. I figured the original person who commented it out wouldn't miss it. I'm also like 99% sure that I was the original person who commented it out and just forgot to delete in a different PR. I checked with myself and I don't mind that it was deleted, so everyone's happy.

### Screenshots/Videos
Before - non legacy: 

![image](https://github.com/user-attachments/assets/15779936-2494-4190-a0a8-205f0249be72)

After - non legacy:

![image](https://github.com/user-attachments/assets/dee5d1f4-d495-4ecf-ab9a-cb436c439f13)

One thing to note is that while the font colour for non legacy is the same for TextStyle.SUMMARY and TextStyle.WINDOW, the shadows are just a touch different - you can see SUMMARY vs WINDOW here below; the SUMMARY is a little more grey, but the stat names in the IV scanner pop up use WINDOW, so this makes the shadows consistent around the board for this:

![image](https://github.com/user-attachments/assets/a55db66b-0f4e-48a9-857b-3696a6ec8ff3)

Before - legacy and nothing better:

![image](https://github.com/user-attachments/assets/e5cba7ce-abd8-4d32-aa80-b242822c6440)

After - legacy and nothing better:

![image](https://github.com/user-attachments/assets/ec0311aa-7cf1-4b51-85a1-0da7848ca04e)

Before - legacy and some worse, some better and some best:

![image](https://github.com/user-attachments/assets/c1f98b6c-2947-474e-8e8d-2fd4345a6fe5)

After - legacy and some worse, some better and some best:

![image](https://github.com/user-attachments/assets/9bbf08a2-9b04-4d6c-ab24-9308b8bb2d28)

## How to test the changes?
[This](https://discord.com/channels/1125469663833370665/1176874654015684739/1274920222910058507) save file has 2 IV scanners on it. You may need to go through your starter select screen and check what you've caught and what IVs you've got, but once you've got an idea you can set the following overrides: 

Line 108 - this changes what pokemon you'll be facing. To set this, use "Species.SPECIESNAME" after the = to make the enemy pokemon that pokemon. For example, to spawn in a charmander, you would do this:

```readonly OPP_SPECIES_OVERRIDE: Species | integer = Species.CHARMANDER;```

Also line 117 lets you change the enemy pokemon's IVs. You can find some IVs from the start screen depending on what pokemon you choose and adjust accordingly. Below is an example:

```readonly OPP_IVS_OVERRIDE: integer | integer[] = [4, 15, 3, 31, 27, 29];```

You should aim to test stats that are worse than what you have, better than what you have (but not max 31 IV) and finally 31 IV, just to check the colour differences

## Checklist
